### DISCUSSION TITLE
How to move a puzzle piece in the middle?

### DISCUSSION THREAD

**Discussion by RukshanJS** [flbulgarelli/headbreaker#45] [Q&A] [answered]
After the puzzle is almost solved, how can we remove a piece in the middle? If I click on a middle piece (surrounded by all 4 sides), then the whole puzzle structure is moved, which is not what is expected.

Expected - remove that selected puzzle piece and be draggable.

**Comment by flbulgarelli** [comment]
Hi!👋 That behavior is actually by design, in order to allow all operations - drag group, connect pieces, push pieces, disconnect pieces - simple in terms of gestures. 

However I know that may not not cover all use cases. For example, we could use the alt or control key to tell that user wants to disconnect the piece - at the cost of a more complex set of user commands. Or perhaps implement a whole new gesture.

But, in any case, that is not supported out of the box. You'll need to implement by yourself at the Konva layer. Here https://github.com/flbulgarelli/headbreaker/discussions/42 there is a code sample that shows how to access the figures - the visual objects on top of the puzzle model. You could try to disconnect the pieces as part of the drag event, on a condition of your preference.

Just ask back if you need more assistance with it! 😁

**Comment by flbulgarelli** [comment]
PS: I have made some progress with integrating those ideas into the core code. See here https://github.com/flbulgarelli/headbreaker/pull/47.

**Comment by flbulgarelli** [comment, marked as answer]
I have just released a new version of `headbreaker` that ships with that functionality. You can check it here: https://flbulgarelli.github.io/headbreaker/#keyboard-gestures. 

You can find more details here: 

https://github.com/flbulgarelli/headbreaker/blob/d01559137fc4819d29314d6948cfe127078a94fa/src/canvas.js#L336-L357

**Reply by JeffroDB** [reply]
Any thoughts on how we could implement this type of functionality -- drag blocks -- if a user is using an iPad or iPhone device which doesn't have a keyboard in play? A very large number of my users are visiting my website via phones or pads.

Perhaps by having a mouse double-click on a group of attached pieces cause them to be treated as a block (another double-click releases the block). I believe in that manner a phone or pad's double tap might do the trick.
